src/Compilers/VisualBasic/Portable/Binding/Binder.vb

```diff
@@ -1067,6 +1067,19 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Return m_containingBinder.QuickAttributeChecker
             End Get
         End Property
+
+        Private Shared Function CheckFeatureAvailability(feature As InternalSyntax.Feature, node As SyntaxNode, diagnostics As DiagnosticBag) As Boolean
+            Dim langVersion As LanguageVersion = DirectCast(node.SyntaxTree.Options, VisualBasicParseOptions).LanguageVersion
+            Dim requiredVersion As LanguageVersion = InternalSyntax.FeatureExtensions.GetLanguageVersion(feature)
```

Comment:
>Dim requiredVersion As LanguageVersion = InternalSyntax.FeatureExtensions.GetLanguageVersion(feature) [](start = 12, length = 101)

It looks like from this line down the code duplicates what ```Friend Shared Function CheckFeatureAvailability(diagnostics As DiagnosticBag, location As Location, languageVersion As LanguageVersion, feature As Feature) As Boolean``` in the Parser does.

Reply:
It does, because I didn't want to make the Binder depend on a static method in the Parser. This is what C# does.

---
In reply to: [217210173](https://github.com/dotnet/roslyn/pull/29724#discussion_r217210173) [](ancestors = 217210173)

Reply:
>  I didn't want to make the Binder depend on a static method in the Parser

This is where we keep the helper and we are using it elsewhere as well. Please delegate to it instead of duplicating the code

---
In reply to: [217217240](https://github.com/dotnet/roslyn/pull/29724#discussion_r217217240) [](ancestors = 217217240,217210173)